web/docs/auth/auth-hooks.md

```diff
@@ -30,6 +32,17 @@ If you are using OAuth, the flow includes extra steps before the signup flow:
   caption="OAuth Flow with Hooks"
 />
 
+<ImgWithCaption
+  source="/img/auth-hooks/login_flow_with_hooks.png"
+  alt="Login Flow with Hooks"
+  caption="Login Flow with Hooks"
+/>
+
+<small>
+
+\* When using the OAuth auth providers, the login hooks are both called before the session is created but the session is created quickly afterward, so it shouldn't make any difference in practice.
```

Comment:
Hm, this _really_ seems to suggest that the comment discussed here should be on the `onAfterLoginHook`: https://github.com/wasp-lang/wasp/pull/2211#discussion_r1699849062

`onBeforeLogin` truly is called before login (session creation), while `onAfterLogin` isn't called after login (session creation), or am I missing something?

Reply:
### Email and username providers

`onBeforeLogin` is called _right before_ session creation.

`onAfterLogin` is called _right after_ session creation.

### OAuth is a bit different

`onBeforeLogin` is called _way before_ session creation i.e. before redirecting to the client with OTC (one time code) which will then exchange the OTC for session (harder to say it's wrong)

`onAfterLogin` is called at the same time (easy to say it's wrong)

### Placement of the comment

That's why I put the comment above the `onBeforeLogin` because it also differs from the email and username cases, so it felt right.

Reply:
Ohh, I see the misunderstanding. You wanted to cover both hooks with the comment above `onBeforeLogin`, and since that one comes before `onAfterLogin`, you put it there.

Ok, makes sense, but I wonder if it's easily missed. Do you think it makes sense to include a note above `onAfterLogin` too? Something simple like: `Check the comment above`.

Reply:
Makes sense 👍  I'll add an extra comment above the `onAfterLogin`.